wolfcrypt/src/ed25519.c

```diff
@@ -60,26 +60,18 @@ static int ed25519_hash(ed25519_key* key, const byte* in, word32 inLen,
     byte* hash)
 {
     int ret;
-    wc_Sha512 sha;
```

Comment:
I like the refactor to move this into ED25519 struct. Perhaps consider wrapping that logic in `WOLFSSL_SMALL_STACK_CACHE` build option like we do elsewhere? That way it won't break any existing ED25519 users memory expectations.

Reply:
Or add a new option for the ED streaming support, since the SHA is required to be in the struct for those... Default streaming on add `WOLFSSL_NO_ED_STREAM` to explicitly disable. `#if !defined(WOLFSSL_NO_ED_STREAM) || defined(WOLFSSL_SMALL_STACK_CACHE)` then put SHA into struct. For the local stack variable use a flow like this:

```
#if !defined(WOLFSSL_NO_ED_STREAM) || defined(WOLFSSL_SMALL_STACK_CACHE)
wc_Sha512* sha = key->she;
#else
wc_Sha512 sha[1];
#endif
```

Reply:
The new arrangement may be better for everyone, won't meaningfully harm any existing users.  The motivation for the refactor, of course, is that the SHA state is actually part of the ED state.  For the streaming API that dependency can't be hidden inside the function the way it can for the one-shot API.  Bigger picture, the SHA state is going to use roughly the same amount of memory whether it's inlined in the ED state struct or put on the stack inside the func call, but inlining it in the ED state means that if the caller puts the ED state in the heap, the SHA state will also be on the heap, which seems to be better behavior.  It doesn't hurt that SHA allocations/deallocations on init/free are now optimized out for the life of the ED key struct, but as it happens., in the default codepath there aren't any.  But if there were, win!